src/main/java/com/poortorich/global/date/util/DateInfoProvider.java

```diff
@@ -0,0 +1,59 @@
+package com.poortorich.global.date.util;
+
+import com.poortorich.global.date.constants.DateConstants;
+import com.poortorich.global.date.domain.MonthInformation;
+import com.poortorich.global.date.domain.WeekInformation;
+import com.poortorich.global.date.domain.YearInformation;
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.Month;
+import java.time.Year;
+import java.time.YearMonth;
+import java.time.temporal.TemporalAdjusters;
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+public class DateInfoProvider {
+
+    public static YearInformation getYearInformation(Year year) {
+        Map<Month, MonthInformation> months = new EnumMap<>(Month.class);
+        Stream.of(Month.values())
+                .forEach(month -> months.put(month, getMonthInformation(year.atMonth(month))));
+
+        return YearInformation.builder()
+                .year(year)
+                .startDate(year.atDay(DateConstants.ONE_DAY))
+                .endDate(year.atMonth(Month.DECEMBER).atEndOfMonth())
+                .months(months)
+                .build();
+    }
+
+    public static MonthInformation getMonthInformation(YearMonth yearMonth) {
+        return MonthInformation.builder()
+                .yearMonth(yearMonth)
+                .startDate(yearMonth.atDay(DateConstants.ONE_DAY))
+                .endDate(yearMonth.atEndOfMonth())
+                .weeks(DateInfoProvider.getWeekInformation(yearMonth))
+                .build();
+    }
+
+    public static List<WeekInformation> getWeekInformation(YearMonth yearMonth) {
```

Comment:
`List<WeekInformation>`을 반환하는데 단수형 네이밍이기 때문에 조금 애매하다고 생각합니다.. 
```suggestion
    public static List<WeekInformation> getWeeksInformation(YearMonth yearMonth) {
```
이런식으로 바꿔보는건 어떨까요? 
```suggestion
    public static List<WeekInformation> getWeeks(YearMonth yearMonth) {
```
이런 것도 괜찮을 거 같습니다.
네이밍의 일관성이 깨진다는 아쉬움은 있지만, 메소드의 하는 일이 더 명확하게 드러나기 때문에 나쁘지 않다고 생각합니다!

Reply:
`getWeeksInformation`으로 수정해볼게요. 의견 감사합니다 😉